honeybee_radiance/modifier/material/bsdf.py

```diff
@@ -74,31 +72,20 @@ class BSDF(Material):
         * is_modifier
         * is_material
     """
-    __slots__ = ('_bsdf_file', '_up_orientation', '_thickness', '_function_file',
-                 '_transform', '_angle_basis', '_front_diffuse_reflectance',
-                 '_back_diffuse_reflectance', '_diffuse_transmittance')
+
+    __slots__ = ('_thickness')
 
     # TODO(): compress file content: https://stackoverflow.com/a/15529390/4394669
     def __init__(self, bsdf_file, identifier=None, up_orientation=None, thickness=0,
                  modifier=None, function_file='.', transform=None, angle_basis=None,
                  dependencies=None):
         """Create BSDF material."""
-        identifier = identifier or '.'.join(os.path.split(bsdf_file)[-1].split('.')[:-1])
-
-        Material.__init__(self, identifier, modifier=modifier,
-                          dependencies=dependencies)
-
-        self.bsdf_file = bsdf_file
-        self.angle_basis = angle_basis
-        self.up_orientation = up_orientation
         self.thickness = thickness or 0
```

Comment:
You should probably move this line after you initiate the class in line 84.

Reply:
Hi @mostaphaRoudsari, I didn't see you added all these comments. I corrected your suggestions except this one, as it will run `_update_values()` when initiating the class, but the thickness attribute is not yet set, which is needed in line 92 in `_update_values()`.